velox/expression/CastHooks.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/CastHooks.h"
+
+namespace facebook::velox::exec {
+
+const std::shared_ptr<CastHooks>& CastHooks::get() {
+  static const std::shared_ptr<CastHooks> instance =
+      std::make_shared<CastHooks>();
+  return instance;
+}
+
+void CastHooks::castTimestampToString(
+    const Timestamp& timestamp,
+    bool legacy,
+    StringWriter<false>& out) {
+  out.copy_from(
```

Comment:
@PHILO-HE I wonder if we can optimize away creation of intermediate strings and copying in the same way you optimizes date format.

Reply:
@mbasmanova, looks we can do the similar optimization here.

Reply:
It seems this optimization needs to be done in `tmToString` function. Can I open an issue for it and make the optimization after this PR? @mbasmanova 
https://github.com/facebookincubator/velox/blob/4466c879770565e0db05b56ee03858d7d95f8417/velox/type/Timestamp.cpp#L237-L240

Reply:
@PHILO-HE @rui-mo Sure. Let's create an issue and optimize this in a separate PR. Thanks.

Reply:
Opened https://github.com/facebookincubator/velox/issues/8067. I will follow up this issue, thanks.